yt/data_objects/static_output.py

```diff
@@ -1872,6 +1879,77 @@ def __eq__(self, other):
     def __hash__(self):
         return hash((self.filename, self.file_id, self.start, self.end))
 
+    def _nonzero_ptf(self, ptf):
+        for ptype, field_list in sorted(ptf.items()):
+            if self.total_particles[ptype] == 0:
+                continue
+            yield ptype, field_list, self.total_particles[ptype]
+
+    def _nonzero_ptypes(self, ptypes=None):
+        if ptypes is None:
+            ptypes = self.total_particles.keys()
+
+        for ptype in sorted(ptypes):
+            if self.total_particles[ptype] == 0:
+                continue
+            yield ptype, self.total_particles[ptype]
+
+    @contextlib.contextmanager
+    def transaction(self, handle=None):
+        # yields an open file handle, without double-reading.
+        if handle is None:
+            with self.open_handle() as handle:
+                yield handle
```

Comment:
ok so I'm experimenting on adding type hints to these methods and I don't have a definitive solution yet, but in any case it seems that rebinding the variable with an `as` isn't making the task any easier so I suggest :
```suggestion
            with self.open_handle() as new_handle:
                yield new_handle
```

Reply:
Ya, I wasn't sure how to approach typing here... but will change the variable name!